### PR TITLE
LPS-87307 Uses SearchParams native API to parse and work with searchParams instead of relying on manual parsing of urls

### PR DESCRIPTION
**Pull request by jbalsas**

### CODE REVIEW

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js [hunk 1]
```diff
@@ -1159,16 +1159,20 @@
 					Util._submitLocked = true;
 				}
 
-				var actionURL = new A.Url(action);
+				var searchParamsIndex = action.indexOf('?');
 
-				var authToken = actionURL.getParameter('p_auth') || '';
+				var searchParams = new URLSearchParams(
+					action.substring(searchParamsIndex)
+				);
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js]: Out of an excess of caution, I think I'd want to make the handling of URLs without a query string a little more explicit (even though we probably don't expect to ever get that kind of URL, and the code still happens to work if you pass it something that isn't actually a search params string).

So maybe something like:

```javascript
var searchParamsIndex = action.indexOf('?');

var searchParams = new URLSearchParams(
  searchParamsIndex !== -1 ?
  action.substring(searchParamsIndex) :
  ''
);
```

File: modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js [hunk 2]
```diff
@@ -1159,16 +1159,20 @@
...
+				var authToken = searchParams.get('p_auth') || '';
 
 				form.append('<input name="p_auth" type="hidden" value="' + authToken + '" />');
 
 				if (authToken) {
-					actionURL.removeParameter('p_auth');
+					searchParams.delete('p_auth');
 
-					action = actionURL.toString();
+					action = action.substring(0, searchParamsIndex + 1) + searchParams.toString();
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/util.js]: I know this is guarded by the `if (authToken)` check but it makes me a little nervous that if this path ever got called — for example, due to a refactoring — when `searchParamsIndex` is `-1`, then `action` ends up being assigned an empty string.

Makes me wonder whether something a bit more verbose would be more robust:

```javascript
var searchParamsIndex = action.indexOf('?');

if (searchParamsIndex === -1) {
  var baseURL = action;
  var queryString = '';
} else {
  var baseURL = action.slice(0, searchParamsIndex);
  var queryString = action.slice(searchParamsIndex + 1);
}
```

then the rest of your code much like before:

```javascript
var searchParams = new URLSearchParams(queryString);

var authToken = searchParams.get('p_auth') || '';

form.append('... etc');

if (authToken) {
  searchParams.delete('p_auth');

  action = baseURL + '?' + searchParams.toString();

  // etc
```

Obviously if we did that, my other comment becomes redundant. The other benefit is that it keeps all the arithmetic related to the index close together at the start.

Having said all that though, I think your changes are correct and would be fine. Let me know what you think.

**Comment by wincent** [comment]
Forwarded upstream here: https://github.com/brianchandotcom/liferay-portal/pull/72606

**Pull request closed without merging**